package/lib/src/cli/commands/resync/resync_command.dart

```diff
@@ -0,0 +1,151 @@
+
+  @override
+  String get name => 'resync';
+
+  @override
+  String get description => 'Resync project'.bold();
+
+  @override
+  Future<void> execute() async {
+    final DataClassPlugin plugin = DataClassPlugin(PhysicalResourceProvider.INSTANCE);
+
+    final DataClassPluginOptions pluginOptions = await DataClassPluginOptions.fromFile(
+        getDataClassPluginOptionsFile(_codeGenerator.directory.path));
+
+    await _codeGenerator.indexProject();
+
+    final List<String> includedPaths = _codeGenerator.registeredFiles;
+
+    plugin.start(WebSocketPluginServer(logger: logger));
+
+    Future<void> generate() async {
+      final AnalysisContextCollection analysis = AnalysisContextCollection(
+        includedPaths: includedPaths,
+        resourceProvider: PhysicalResourceProvider.INSTANCE,
+      );
+
+      await plugin.handleAnalysisSetContextRoots(
+        AnalysisSetContextRootsParams(<ContextRoot>[
+          for (final String path in includedPaths) ContextRoot(path, <String>[]),
+        ]),
+      );
+
+      for (final String path in includedPaths) {
+        final io.File file = io.File(path);
+
+        final AssistRequest request = await plugin.getAssistRequest(
+          EditGetAssistsParams(path, 0, file.lengthSync()),
+        );
+
+        final _SimpleAssistCollector collector = _SimpleAssistCollector();
+        final DataClassAssistContributor dataClassContributor =
+            DataClassAssistContributor(path, pluginOptions: pluginOptions);
+
+        final ResolvedUnitResult resolvedUnitResult = await analysis
+            .contextFor(path)
+            .currentSession
+            .getResolvedUnit(path)
+            .then((SomeResolvedUnitResult value) => value as ResolvedUnitResult);
+
+        await dataClassContributor.computeAssists(
+          _SimpleAssistRequest(
+            length: request.length,
+            offset: request.offset,
+            resourceProvider: request.resourceProvider,
+            result: resolvedUnitResult,
+          ),
+          collector,
+        );
+
+        String content = file.readAsStringSync();
+        for (final PrioritizedSourceChange assist in collector.assists) {
+          for (final SourceFileEdit sourceFileEdit in assist.change.edits) {
+            for (final SourceEdit edit in sourceFileEdit.edits) {
+              content = content.replaceRange(edit.offset, edit.end, edit.replacement);
+            }
+          }
+        }
+        file.writeAsStringSync(DartFormatter(
+          pageWidth: pluginOptions.generatedFileLineLength,
+        ).format(content));
+      }
+
+      await _codeGenerator.indexProject(forceClear: true);
+      await _codeGenerator.buildProject();
+    }
+
+    await generate();
+
+    // a second generate is necessary as code generation might depend on code which needs to be generated
+    // example: super classes with DataClass annotation
+    await generate();
+
+    logger.info('~ Resync completed');
```

Comment:
I think it would be nice to log the total duration of the resync command.

Reply:
Added

Reply:
Added 🙏